File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0 or (x % 10 == 0 and x != 0):
            return False

        reverted_second_half = 0
        remaining_first_half = x

        while remaining_first_half > reverted_second_half:
            last_digit = remaining_first_half % 10
            reverted_second_half = reverted_second_half * 10 + last_digit
            remaining_first_half //= 10
        return remaining_first_half == reverted_second_half or remaining_first_half == reverted_second_half // 10

File: 0009-palindrome-number/0009-palindrome-number.go
package main

import "fmt"

func isPalindrome(x int) bool {
	if x < 0 || (x%10 == 0 && x != 0) {
		return false
	}

	var rev int = 0
	for x > rev {
		last_digit := x % 10
		rev = rev*10 + last_digit
		x /= 10
	}
	return x == rev || x == rev/10
}